tensorflow/core/util/gpu_device_functions.h

```diff
@@ -783,7 +822,7 @@ CREATE_CUDA_DEVICE_FUNCTION_ALIAS(GpuAtomicAdd, CudaAtomicAdd);
 // GpuAtomicSub
 template <typename T, typename U>
 __device__ detail::ToTypeIfConvertible<U, T> GpuAtomicSub(T* ptr, U value) {
-  return atomicSub(ptr, value);
+  return atomicSub(detail::AddressSpaceHint<1>(ptr), value);
```

Comment:
may I ask why `atomicAdd(detail::AddressSpaceHint<3>(dst), val);` is using shared memory while `atomicSub(detail::AddressSpaceHint<1>(ptr), value);` is using global memory?

Reply:
AddressSpaceHint<3> is only for GpuAtomicAddShared that is used by rocm specific kernels.

Reply:
Looks like there is no `GpuAtomicSubShared`, which would use addr space 3 if it existed. I'm not sure why it doesn't exist since sub just calls add anyway.